sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -259,8 +259,26 @@ internal static bool TryExtractErrorContent(Response response, out ResponseError
                     return false;
                 }
                 // Try the ErrorResponse format and fallback to the ResponseError format.
+
+#if NET6_0_OR_GREATER
+                var deserializedErrorResponse = System.Text.Json.JsonSerializer.Deserialize(content, typeof(ErrorResponse), ResponseErrorSourceGenerationContext.Default);
+                if (deserializedErrorResponse != null)
+                {
+                    var errorResponse = (ErrorResponse)deserializedErrorResponse;
+                    error = errorResponse?.Error;
+                }
+                if (error == null)
+                {
+                    var deserializedResponseError = System.Text.Json.JsonSerializer.Deserialize(content, typeof(ResponseError), ResponseErrorSourceGenerationContext.Default);
+                    if (deserializedResponseError != null)
+                    {
+                        error = (ResponseError)deserializedResponseError;
+                    }
+                }
```

Comment:
Why can't this code be written like the code in the `#else` section?

```suggestion
                error = System.Text.Json.JsonSerializer.Deserialize<ErrorResponse>(content, ResponseErrorSourceGenerationContext.Default.ErrorResponse)?.Error;
                error ?? = System.Text.Json.JsonSerializer.Deserialize<ResponseError>(content, ResponseErrorSourceGenerationContext.Default.ResponseError);
```

Reply:
I can try it again, it didn't look like the generic version has an overload that takes in a JsonSerializerContext in .NET 6 - https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=net-6.0 so it wasn't building

Reply:
This should be the one you will want to use:

https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=net-6.0#system-text-json-jsonserializer-deserialize-1(system-readonlyspan((system-char))-system-text-json-serialization-metadata-jsontypeinfo((-0)))

```
public static TValue? Deserialize<TValue> (ReadOnlySpan<char> json, System.Text.Json.Serialization.Metadata.JsonTypeInfo<TValue> jsonTypeInfo);
```

Reply:
Ahh ok I see, thanks!